### ISSUE TITLE
Problems with addon setings

### ISSUE THREAD

**Issue by radekrudak** [albinaask/Log#6]
# Issue
I had another issue with setting defaulting to nil but that was fixed with godot restart.

How ever this one didn't passed away after simple restarting editor.

In log-stream.gd in function _internal_log on line 119
 there's ProjectSettings.get_setting(Settings.LOG_MESSAGE_FORMAT_KEY) which for some reason defaults to nil, despite fact that
in project setting i can see default being {log_name}/{level} [lb]{hour}:{minute}:{second}[rb] {message}

when i change LOG_MESSAGE_FORMAT_KEY aka "addons/logger/log_message_format" said setting problem disappears ( didn't test if any of library functionality works tho.)

Restarting editor and even reinstalling plugins didn't help 

now changed after changing setting a bit and restarting editor, problem has been fixed

Now it seems to work, but reverting setting "addons/logger/log_message_format"  to default sets them to null just like reverting
icon setting in addon setting reverts to null.

On top of that order of settings seems to change from time to time.

It's not a huge problem but it's off putting when you want to try this plugin for first time, and annoying, and since i cannot say
certainly since i don't have experience in godot plugins it seems like there's problem in how plugin handles it's settings. 

# Temporary Fix
Add default values on ProjectSettings.get_setting calls or at least this particular call i haven't tested it but 
it seems like for some reason call cannot find setting so it defaults to default set by argument, which defaults to null

# Steps to reproduce
Install plugin, i guess.

# Platform 
Linux

**Comment by albinaask** [comment]
Thank you for the report and I will defiantly fix it as soon as I can. There has been a lot going on in my life for a while, so Log has been put on the shelf for some time unfortunately.

**Comment by LaurentOngaro** [comment]
Same error for me, I found a fast and dirty  fix by editing the `res://addons/logger/log-stream.gd` file and replacing the #119 line
```
var msg = String(ProjectSettings.get_setting(Settings.LOG_MESSAGE_FORMAT_KEY)).format(format_data)
```

by

```
	var setting:=ProjectSettings.get_setting(Settings.LOG_MESSAGE_FORMAT_KEY)
	if not setting:
		setting= "{log_name}/{level} [lb]{hour}:{minute}:{second}[rb] {message}"

	var msg = String(setting).format(format_data)
```

**Comment by albinaask** [comment]
I think this should also be fixed in 1.1.2 that is just released. Please let me know if this fixes your problems or not. The issues I've been able to recreate, I've fixed. There is also a [bug with Godot](https://github.com/godotengine/godot/issues/87911) that I found during testing that may be causing other issues.

**Comment by albinaask** [comment]
No worries mate!
